drivers/video/fbdev/rm68090fb.c

```diff
@@ -0,0 +1,1124 @@
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option)any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+ */
+#include <linux/cdev.h>
+#include <linux/module.h>
+#include <linux/kernel.h>
+#include <linux/fb.h>
+#include <linux/dma-mapping.h>
+#include <linux/device.h>
+#include <linux/platform_device.h>
+#include <linux/uaccess.h>
+#include <linux/pm_runtime.h>
+#include <linux/interrupt.h>
+#include <linux/wait.h>
+#include <linux/clk.h>
+#include <linux/cpufreq.h>
+#include <linux/console.h>
+#include <linux/spinlock.h>
+#include <linux/slab.h>
+#include <linux/delay.h>
+#include <linux/lcm.h>
+#include <linux/clk-provider.h>
+#include <video/of_display_timing.h>
+#include <linux/gpio.h>
+#include <linux/omapfb.h>
+#include <linux/compiler.h>
+#include <linux/workqueue.h>
+#include <asm/io.h>
+#include <asm/gpio.h>
+#include <asm/arch-suniv/cpu.h>
+#include <asm/arch-suniv/dma.h>
+#include <asm/arch-suniv/gpio.h>
+#include <asm/arch-suniv/intc.h>
+#include <asm/arch-suniv/lcdc.h>
+#include <asm/arch-suniv/debe.h>
+#include <asm/arch-suniv/clock.h>
+#include <asm/arch-suniv/common.h>
+
+#define MIYOO_FB0_PUT_OSD     _IOWR(0x100, 0, unsigned long)
+#define MIYOO_FB0_SET_MODE    _IOWR(0x101, 0, unsigned long)
+#define MIYOO_FB0_GET_VER     _IOWR(0x102, 0, unsigned long)
+#define MIYOO_FB0_SET_FLIP    _IOWR(0x103, 0, unsigned long)
+#define MIYOO_FB0_SET_FPBP    _IOWR(0x104, 0, unsigned long)
+#define MIYOO_FB0_GET_FPBP    _IOWR(0x105, 0, unsigned long)
+
+#define NO_POCKET_GO             1
+#define MIYOO3                1
+
+#define LRAM_NUM              1
+#define PALETTE_SIZE          256
+#define DRIVER_NAME           "miyoofb"
+#define SLCD_RESET		        ((32 * 4) + 11) // PE11
+#define SLCD_TE		  		      ((32 * 4) + 10) // PE10
+
+static bool flip=false;
+module_param(flip,bool,0660);
+
+// Which LCD controller are we driving? E.g. R61520, R61526, R61581, etc. 
+// Should be a number 1-4, but I'm not yet sure which numbers correspond to which panels
+// 1 = R61520 - R61520 doesn't support the 0x04 read_DDB_start / Read Display ID Information command.
+// 2 = ST7789S
+// 3 = R61505W
+// 4 = ?
+// Will be automatically detected if left unset (which usually works)
+static uint32_t version=0;
+static bool invert=false;
+module_param(version,uint,0660);
+module_param(invert,bool,0660);
```

Comment:
We probably don't need the version stuff now that we have separate drivers for separate panels. Although it's not going to be as easy as just deleting it; there's some code below that still checks both this and the detected version. Doesn't necessarily need to be changed in this PR, but it wouldn't hurt.